src/fs.rs

```diff
@@ -440,6 +440,8 @@ impl Stream for ChunkedReadFile {
                 let max_bytes: usize;
                 max_bytes = cmp::min(size.saturating_sub(counter), 65_536) as usize;
                 let mut buf = Vec::with_capacity(max_bytes);
+                // safe because memory is initialized/overwritten immediately
+                unsafe { buf.set_len(max_bytes); }
```

Comment:
This sets the length of the buffer to `max_bytes`, but further down writes `nbytes` and `nbytes ≤ max_bytes`. So, in `Bytes::from(buf)` you return a buffer that may contain `max_bytes - nbytes` uninitialized bytes. Am I reading this correctly?

Reply:
You're right.

Reply:
You could add `debug_assert!(nbytes <= max_bytes)` if you really wanted to make sure without any cost in release mode but it'd only test that `std::fs` is implemented correctly…

Reply:
I think it would make a little difference to just initialize vec like
```rust
let mut buf = vec![0; max_bytes]
```
?

Though personally I don't see an issue, I wonder why you don't want just zero initialize it?

By the way I think in this case we could use `BytesMut` directly and then freeze it into `Bytes` directly if you're up for improvement of this code

Reply:
The main reason I used set_len is that (I assume) this was the original intent of the code, with minimal changes.

I see no point in initializing to zero here, that's unnecessarily polluting the cache and using up memory bandwidth in a place where that might actually make a difference.

I considered using truncate instead of the second set_len, but after looking at [its O(n) implementation](https://doc.rust-lang.org/src/alloc/vec.rs.html#695-706) I decided against that. 

I'm not opposed to using BytesMut, but unless I missed something in the API, we can't avoid the unsafe set_len there, either (please correct me if I'm wrong).

Reply:
Well I'm fine with it, btw `BytesMut` is mostly to avoid need for conversion from Vec to Bytes but I see your reasoning and it is fine by me